packages/react-core/src/demos/ComposableMenu/examples/ComposableTypeaheadSelect.tsx

```diff
@@ -0,0 +1,195 @@
+        } else {
+          indexToFocus = focusedItemIndex + 1;
+        }
+      }
+
+      setFocusedItemIndex(indexToFocus);
+    }
+  };
+
+  const onInputKeyDown = (event: React.KeyboardEvent<HTMLInputElement>) => {
+    const enabledMenuItems = menuItems.filter(menuItem => !menuItem.isDisabled);
+    const [firstMenuItem] = enabledMenuItems;
+    const focusedItem = focusedItemIndex ? enabledMenuItems[focusedItemIndex] : firstMenuItem;
+
+    switch (event.key) {
+      // Select the first available option
+      case 'Enter':
+        if (isMenuOpen) {
+          setInputValue(String(focusedItem.children));
+          setIsSelected(true);
+        }
+
+        setIsMenuOpen(prevIsOpen => !prevIsOpen);
+        setFocusedItemIndex(null);
+        focusOnInput();
+
+        break;
+      case 'Tab':
+      case 'Escape':
+        setIsMenuOpen(false);
+        break;
+      case 'ArrowUp':
+      case 'ArrowDown':
+        handleMenuArrowKeys(event.key);
+        break;
+      default:
+        !isMenuOpen && setIsMenuOpen(true);
+    }
+  };
+
+  // Close the menu when a click occurs outside of the menu toggle content
+  const onDocumentClick = (event: MouseEvent) => {
+    if (!menuToggleRef.current?.contains(event?.target as HTMLElement)) {
+      setIsMenuOpen(false);
+    }
+  };
+
+  const toggleMenuOpen = () => {
+    setIsMenuOpen(prevIsOpen => !prevIsOpen);
+    textInputRef.current?.focus();
+  };
+
+  const onTextInputChange = (value: string) => {
+    setInputValue(value);
+    setIsSelected(false);
+  };
+
+  return (
+    <Popper
+      trigger={
+        <MenuToggle variant="typeahead" onClick={toggleMenuOpen} innerRef={menuToggleRef} isFullWidth>
+          <TextInputGroup isPlain>
+            <TextInputGroupMain
+              value={inputValue}
+              onClick={toggleMenuOpen}
+              onChange={onTextInputChange}
+              onKeyDown={onInputKeyDown}
+              id="typeahead-select-input"
+              autoComplete="off"
+              innerRef={textInputRef}
+            />
+
+            <TextInputGroupUtilities>
+              {!!inputValue && (
+                <Button variant="plain" onClick={() => setInputValue('')} aria-label="Clear input value">
+                  <TimesIcon />
+                </Button>
+              )}
+              <Button variant="plain" onClick={toggleMenuOpen} aria-label="Toggle menu">
+                <CaretDownIcon />
+              </Button>
+            </TextInputGroupUtilities>
+          </TextInputGroup>
+        </MenuToggle>
+      }
+      popper={
+        <Menu id="select-menu" onSelect={onMenuSelect} selected={isSelected && inputValue}>
+          <MenuContent>
+            <MenuList>
+              {menuItems.map((itemProps, index) => (
+                <MenuItem
+                  key={itemProps.itemId || itemProps.children}
+                  className={css(focusedItemIndex === index && styles.modifiers.focus, itemProps.className)}
```

Comment:
Would it be worth introducing a property flag for `MenuItem` rather than having the user manually toggle the css through the custom classname? Not a blocker though as I'm good either way.

Reply:
I don't think consumers are very used to handling our `styles` object to get classnames, so it probably is more usable to add an 'isFocused' prop.

Reply:
That's a good point. I've addressed this in the latest commit.